x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -238,52 +238,54 @@ private static boolean checkGroupByOrder(LogicalPlan p, Set<Failure> localFailur
             Set<LogicalPlan> groupingFailures, Map<String, Function> functions) {
         if (p instanceof OrderBy) {
             OrderBy o = (OrderBy) p;
-            if (o.child() instanceof Aggregate) {
-                Aggregate a = (Aggregate) o.child();
 
-                Map<Expression, Node<?>> missing = new LinkedHashMap<>();
-                o.order().forEach(oe -> {
-                    Expression e = oe.child();
-                    // cannot order by aggregates (not supported by composite)
-                    if (Functions.isAggregate(e)) {
-                        missing.put(e, oe);
-                        return;
-                    }
+            o.forEachDown(child -> {
```

Comment:
I would limit the down checks since in the future, with subqueries there might multiple `Aggs` underneath.
In other words an order by can have a Filter/Project&Filter before having an Agg. Checking the entire hierarchy seems too broad...